doctoc.js

```diff
@@ -4,13 +4,14 @@
 
 var path      =  require('path')
   , fs        =  require('fs')
+  , os        =  require('os')
   , minimist  =  require('minimist')
   , file      =  require('./lib/file')
   , transform =  require('./lib/transform')
   , files;
 
 function cleanPath(path) {
-  var homeExpanded = (path.indexOf('~') === 0) ? process.env.HOME + path.substr(1) : path;
+  var homeExpanded = (path.indexOf('~') === 0) ? path.join(os.homedir() + path.substr(1)) : path;
```

Comment:
While using `os.homedir()` indeed improves cross-platform compatibility, I noticed that the introduction of `path.join()` deviates slightly from the existing style. I'm unsure if this stylistic change was intended for this PR.

Additionally, if `path.join()` is used, it should accept multiple arguments (e.g., `path.join(os.homedir(), path.substr(1))`) rather than a single concatenated string.

Let me know your thoughts on this!

Reply:
Agreed! Embarrassingly, when I first tested these changes I must have forgotten to quote:

```console
$ ./doctoc.js ~/Documents/...
```

```console
$ ./doctoc.js '~/Documents/...'
```

I re-ran and it works as expected. I do prefer `path.join` over concatenation - I fixed up the code to allow for that (renaming the parameter in the function).